apis/python/tests/test_basic_anndata_io.py

```diff
@@ -1022,44 +1024,68 @@ def test_uns_io(tmp_path, outgest_uns_keys):
         uns=uns,
         dtype=X.dtype,
     )
-    original = adata.copy()
+    adata0 = deepcopy(adata)
 
     soma_uri = tmp_path.as_posix()
 
     tiledbsoma.io.from_anndata(soma_uri, adata, measurement_name="RNA")
-    verify_obs_and_var_eq(original, adata)
+
+    # NOTE: `from_anndata` mutates user-provided DataFrames in `uns`, demoting their index to a column named "index",
```

Comment:
If `from_anndata` is mutating its caller's AnnData object, it must not.

It can mutate its own copy all it wants.

Reply:
This PR documents and verifies existing behavior.

#2874 builds on this PR, fixing the `uns`-mutation problem.